### DISCUSSION TITLE
How to manage and save/read sensitive data with Strapi?

### DISCUSSION THREAD

**Discussion by biapar** [strapi/strapi#7690] [Legacy Topics]
How to manage and save/read sensitive data with Strapi?
How safe is Strapi secure to manage this type of data?

**Comment by derrickmehaffy** [comment]
Can you please clarify what you mean?

**Comment by biapar** [comment]
I’ve to manage sensitive data that must be encrypted into the db and must be not saved in clear format.

**Reply by derrickmehaffy** [reply]
In this case you will want to override the services (find, findOne, create, update) or use the lifecycle callbacks (which is better as they fire for both Admin and REST/GraphQL)

https://strapi.io/documentation/v3.x/concepts/models.html#lifecycle-hooks

**Reply by biapar** [reply]
Data must be saved encrypted into the database
The manager can see all data from the panel
The customer support only some data and fileds.

**Reply by derrickmehaffy** [reply]
Yes @biapar you would use for example:

- beforeCreate
- beforeUpdate

To encrypted the data into the database, then for fetching you would use:

- afterFind
- afterFindOne

To decrypt the data, we do not have a built in option to handle this automatically, so it's something you will need to add to your own project.
